cdm/src/main/java/ucar/nc2/ft2/simpgeometry/SimpleGeometryFeature.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 1998-2018 John Caron and University Corporation for Atmospheric Research/Unidata
+ * See LICENSE for license information.
+ */
+package ucar.nc2.ft2.simpgeometry;
+
+import ucar.ma2.DataType;
+import ucar.ma2.InvalidRangeException;
+import ucar.nc2.Attribute;
+import ucar.nc2.AttributeContainerHelper;
+import ucar.nc2.Dimension;
+import ucar.nc2.VariableSimpleIF;
+import ucar.nc2.dataset.CoordinateAxis;
+import ucar.nc2.ft2.simpgeometry.adapter.SimpleGeometryCS;
+import ucar.nc2.util.Indent;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Formatter;
+import java.util.List;
+
+/**
+ * SimpleGeometry - forked from Coverage.java
+ * Immutable after setCoordSys() is called.
+ *
+ * @author Katie
+ * @author Caron
+ * @since 8/13/2018
+ */
+// @Immutable
+public class SimpleGeometryFeature implements VariableSimpleIF{
+  private final String name;
+  private final DataType dataType;
+  private final AttributeContainerHelper atts;
+  private final String units, description;
+  private final String coordSysName;
+  protected final Object user;
+  
+  private int[] shapes;
+  private CoordinateAxis xAxis, yAxis, zAxis, IDAxis;
+  
+  private final GeometryType geometryType; // use enum
+
+  private SimpleGeometryCS coordSys; // almost immutable use coordsys
+
+  public SimpleGeometryFeature(String name, DataType dataType, List<Attribute> atts, String coordSysName, String units, String description, Object user, GeometryType geometryType) {

```

Comment:
Since this is in the cdm package and public, could we get a doc string?

Reply:
fixed